src/routes/event/add.tsx

```diff
@@ -18,48 +20,55 @@ export default function EventAdd() {
       }
     });
   }, []);
-  const [location, setLocation] = React.useState<string | null>(null);
-  const [category, setCategory] = React.useState<string | null>(null);
+  const [locationOptions, setLocationOptions] = React.useState<string[]>([]);
+  const [categoryOptions, setCategoryOptions] = React.useState<string[]>([]);
+  const [eventStartDate, setEventStartDate] = React.useState<Date | null>(null);
+  const [eventEndDate, setEventEndDate] = React.useState<Date | null>(null);
 
   const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     const data = new FormData(event.currentTarget);
-    const eventName = data.get("eventName") as string;
-    const eventDescription = data.get("eventDescription") as string;
-    const eventStartDate = data.get("eventStartDate") as string;
-    const eventEndDate = data.get("eventEndDate") as string;
-    const eventLocation = data.get("eventLocation") as string;
-    const eventCategory = data.get("eventCategory") as string;
-    const newEventKey = push(ref(database, "events")).key;
-    if (newEventKey === null) {
-      return;
-    }
-    const eventNameData = {
-      [newEventKey]: eventName,
+    const eventData = {
+      eventName: data.get("eventName") as string,
+      eventDescription: data.get("eventDescription") as string,
+      eventStartDate: eventStartDate?.toISOString().split("T")[0] as string,
+      eventEndDate: eventEndDate?.toISOString().split("T")[0] as string,
+      eventLocation: data.get("eventLocation") as string,
+      eventCategory: data.get("eventCategory") as string,
     };
-    const eventDetailsData = {
-      eventDescription: eventDescription,
-      eventOrganizer: auth.currentUser?.uid,
-      eventStartDate: eventStartDate,
-      eventEndDate: eventEndDate,
-      eventLocation: eventLocation,
-      eventCategory: eventCategory,
-    };
-    const categoryData = {
-      [newEventKey]: true,
-    };
-    const locationData = {
-      [newEventKey]: true,
-    };
-    set(ref(database, "event-details/" + newEventKey), eventDetailsData);
-    update(ref(database, "event-names"), eventNameData);
-    update(ref(database, "locations/" + eventLocation), locationData);
-    if (eventCategory !== null && eventCategory !== "") {
-      update(ref(database, "categories/" + eventCategory), categoryData);
-    }
-    navigate("/event/list");
+    addEvent(eventData);
+    navigate("/");
   };
 
+  useEffect(() => {
+    get(ref(database, "location-names")).then((snapshot) => {
+      if (snapshot.exists()) {
+        const locationsResponseData = snapshot.val();
+        const newLocationOptions: string[] = [];
+        Object.keys(locationsResponseData).forEach((key) => {
+          newLocationOptions.push(key);
+        });
+        setLocationOptions(newLocationOptions);
+      } else {
+        setLocationOptions([]);
+      }
```

Comment:
this is pointless, its already an emrty array on creation, this will only call it once, and will cause a re-render as you are giving it a new ref to a new emprty array.